custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,221 @@
+                    self.update_temp(temp_sensor_state)
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
+
+        self.schedule_update_ha_state(True)
+
+    async def _async_temp_sensor_changed(self, entity_id, old_state, new_state) -> None:
+        """Handle temperature changes."""
+        if new_state is None or old_state == new_state:
+            return
+
+        self.update_temp(new_state)
+        self.schedule_update_ha_state()
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_ON_OFF in self.device.states:
+            if (
+                self.executor.select_state(OverkizState.CORE_ON_OFF)
+                == OverkizCommandParam.OFF
+            ):
+                return HVAC_MODE_OFF
+
+        if (
+            OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE
+            in self.device.states
+        ):
+            return TAHOMA_TO_HVAC_MODES[
+                self.executor.select_state(
+                    OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_OPERATING_MODE
+                )
+            ]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        await self.executor.async_execute_command(
+            "setActiveMode", HVAC_MODES_TO_TAHOMA[hvac_mode]
+        )
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return TAHOMA_TO_PRESET_MODES[
+            self.executor.select_state(
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_SETPOINT_MODE
+            )
+        ]
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        await self.executor.async_execute_command(
+            "setManuAndSetPointModes", PRESET_MODES_TO_TAHOMA[preset_mode]
+        )
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        current_operation = self.executor.select_state(
+            "ovp:HeatingTemperatureInterfaceOperatingModeState"
+        )
+
+        if current_operation in TAHOMA_TO_HVAC_ACTION:
+            return TAHOMA_TO_HVAC_ACTION[current_operation]
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            if self.preset_mode == PRESET_NONE:
```

Comment:
Why is the target temperature `None` when it is in auto mode? Is that how it works with a heater?

Reply:
So this is the interesting part. This is an "ambiance thermostat".
It has 2 working modes:
- Auto:  Run the planned programmed scenario (ex: eco from 23H00 - 06H00 and comfort from 06H00 - 23H00), so it's the HVAC_MODE_AUTO behavior
- Manu: Run the heating with the current setting (ex: manu - eco = 17 deg, manu - comfort : 19 deg)

When you change the temperature setting in manual mode, it changes the temperature used in the auto (ex: manu - eco set at 18 deg, then in auto when it's the eco's time the temperature is set to 18 deg).

Here's the manual of one of the numerous variation (couldn't find an English version) https://damrexelprod.blob.core.windows.net/medias/2fb79f22-2075-4f5a-93d3-112b14a4b8c2)

There's also the cooling mode which reverse how it works (open the valve when over the set temperature, close if under), it's used in heated/cooled floor for exemple.